MCprep_addon/spawner/mcmodel.py

```diff
@@ -372,22 +395,57 @@ def add_model(
 			)
 
 			face.normal_update()
-			
+
 			# Give slight offset by normal for overlay geometry
 			if face_mat == "#overlay":
 				bmesh.ops.translate(bm, verts=face.verts,
-						    vec=0.02 * face.normal)
+									vec=0.0025 * face.normal)
 
 			for j in range(len(face.loops)):
 				# uv coords order is determened by the rotation of the uv,
 				# e.g. if the uv is rotated by 180 degrees, the first index
 				# will be 2 then 3, 0, 1.
 				face.loops[j][uv_layer].uv = uvs[(j + uv_idx) % len(uvs)]
 
-			# Assign the material on face
-			if face_mat is not None and face_mat in materials:
-				face.material_index = materials.index(face_mat)
+			# Using materials_remap to remap the index, used for the block with remapping "#side"
+			# Stored material index for getting the texture
+			material_index = 0
+			if face_mat is not None and (face_mat in materials or face_mat in materials_remap):
```

Comment:
This could be reduced to the following
```suggestion
			if face_mat and (face_mat in materials or face_mat in materials_remap):
```